### PR TITLE
elemental-airgap: add support to OS images

### PR DESCRIPTION
**Pull request by fgiudici**
Allow to extract and download the OS images specified in the channel image shipped with the Elemental Operator chart.
Build a new channel image targeting the local registry of the airgapped scenario (which is now a mandatory option).

Fixes https://github.com/rancher/elemental/issues/950

### CODE REVIEW

**Comment by fgiudici** [comment]
> LGTM
> 
> However I am wondering if another approach wouldn't be better. What if we define a repository field in channel CR and then make use of it when creating the ManagedOSVersions? So this way we could eventually list images as:
> 
> ```
> rancher/elemental-teal/5.4:1.2.2
> rancher/elemental-teal/5.3:1.1.5
> ...
> ```
> 
> And use a custom repository field from the channel to actually compute the full image reference for the image os version. This way we could eventually omit having to recreate the channel image and just reuse it.
> 
> Probably such a value should not be part of the helm chart values, but just a custom field of the channel resource itself. Something like that should really easy to implement.

Yeah, I was thinking something similar. We may want to use the registryUrl from the helm chart, that we may pass to the operator binary, as that already hints that we are using a custom registry, likely an airgap scenario.
Anyway, it is also true that the channel by definition binds specific OS images... shouldn't a custom deployment create its own channel? But yes, if they just want to airgap the official one, it could be handy to keep the original container (also for following updates). 
Anyway, the issue is that we have to unpack the channel to extract the OS images to mirror, so the OS airgap story will not be trivial in any case 🤔 
Last point to consider: airgap in the Rancher Marketplace. The OS images will not be mirrored there, so we will need to instruct the user to create its own OS channel, likely adding options to this script to just do the channel part when needed.
So, long story short: I agree with you, we have to improve the airgap story for the channel/OS part, this is just a first step 👍🏼

**Comment by davidcassany** [comment]
> Anyway, the issue is that we have to unpack the channel to extract the OS images to mirror, so the OS airgap story will not be trivial in any case 🤔

Yes indeed, I believe we should not try to make OS images airgap story fully automated, specially if we embrace the idea of derivatives in some or other way. The thing is that it would be nice having a way to actually redirect the channel to some mirrored (or even modified) images. Someone could automate the mirroring (ro derivatives generation) based on the release channel input. Any way I agree this is all food for later :+1:
